### PR TITLE
Add troubleshooting guide for corrupt repository

### PR DESCRIPTION
**Pull request by andreidan**
This adds a troubleshooting guide for the situation when multiple clusters
are writing to the same snapshot repository, resulting in a corrupt repository.

Preview link https://elasticsearch_88391.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/add-repository.html

### CODE REVIEW

File: docs/reference/troubleshooting/snapshot/add-repository.asciidoc [hunk 1]
```diff
@@ -0,0 +1,11 @@
+[[add-repository]]
+== Mark repository as read-only in other deployments and re-create it in this deployment
```
Comment (andreidan) [inline docs/reference/troubleshooting/snapshot/add-repository.asciidoc:2]: I don't like this title. Any better suggestions?

Comment (lockewritesdocs) [reply docs/reference/troubleshooting/snapshot/add-repository.asciidoc:2]: In troubleshooting pages, the title typically denotes the problem that the user is encountering and trying to solve. In which case, maybe something like _Multiple deployments writing to the same repository_ could be a good title.

File: docs/reference/tab-widgets/troubleshooting/snapshot/corrupt-repository.asciidoc [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+
+Note that now we're configuring the main (current) deployment.
+
+. Open the main deployment's side navigation menu (placed under the Elastic logo in the upper left corner)
+and go to **Stack Management > Snapshot and Restore > Repositories**.
++
+[role="screenshot"]
+image::images/repositories.png[{kib} Console,align="center"]
+
+. Get the details for the repository we'll readd by clicking on the repository
```
Comment (dakrone) [inline docs/reference/tab-widgets/troubleshooting/snapshot/corrupt-repository.asciidoc:52]:
```suggestion
. Get the details for the repository to re-add by clicking on the repository
```

File: docs/reference/tab-widgets/troubleshooting/snapshot/corrupt-repository.asciidoc [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+. Select "Verify repository" to confirm that your settings are correct and the 
+deployment can connect to your repository. 
+//end::kibana-api-ex[]
+// end::cloud[]
+
+// tag::self-managed[]
+Fixing the corrupted repository will entail making changes in multiple clusters
+that write to the same snapshot repository.
+Only one cluster must be writing to a repository. Let's call the cluster 
+we want to keep writing to the repository the "main" cluster (the current cluster),
```
Comment (dakrone) [inline docs/reference/tab-widgets/troubleshooting/snapshot/corrupt-repository.asciidoc:89]:
```suggestion
we want to keep writing to the repository the "primary" cluster (the current cluster), 
```

File: docs/reference/tab-widgets/troubleshooting/snapshot/corrupt-repository.asciidoc [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+    }
+  }
+}
+----
+// TESTRESPONSE[skip:the result is for illustrating purposes only]
++
+<1> Represents the current configuration for the repository.
+
+. Using the settings retrieve above let's add the `readonly: true` option to mark
+it as read-only:
```
Comment (dakrone) [inline docs/reference/tab-widgets/troubleshooting/snapshot/corrupt-repository.asciidoc:123]:
```suggestion
. Using the settings retrieved above, add the `readonly: true` option to mark
it as read-only:
```

File: docs/reference/tab-widgets/troubleshooting/snapshot/corrupt-repository.asciidoc [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
++
+[source,console-result]
+------------------------------------------------------------------------------
+{
+  "acknowledged": true
+}
+------------------------------------------------------------------------------
+// TESTRESPONSE[skip:the result is for illustrating purposes only]
+
+At this point, it's only the main (current) cluster that has the repository marked
```
Comment (dakrone) [inline docs/reference/tab-widgets/troubleshooting/snapshot/corrupt-repository.asciidoc:160]:
```suggestion
At this point, it's only the primary (current) cluster that has the repository marked
```

File: docs/reference/tab-widgets/troubleshooting/snapshot/corrupt-repository.asciidoc [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+as writeable.
+{es} sees it as corrupt though so let's remove the repository and re-add it so that 
+{es} can resume using it:
+
+Note that now we're configuring the main (current) cluster.
```
Comment (dakrone) [inline docs/reference/tab-widgets/troubleshooting/snapshot/corrupt-repository.asciidoc:165]:
```suggestion
Note that now we're configuring the primary (current) cluster.
```

**Review by dakrone — COMMENTED** [review]
I left some general comments, I think we should replace "re-add" with "recreate" since re-add is an awkward semi-not-really-a-word-word.

Otherwise, this is a semi-complicated remediation, so I'm not sure how to simplify it more.

File: docs/reference/troubleshooting/snapshot/add-repository.asciidoc (hunk 1 above)
Comment (andreidan) [reply docs/reference/troubleshooting/snapshot/add-repository.asciidoc:2]: Thanks for the suggestion Adam. I'll make this change.

File: docs/reference/tab-widgets/troubleshooting/snapshot/corrupt-repository.asciidoc [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+and go to **Stack Management > Snapshot and Restore > Repositories**.
++
+[role="screenshot"]
+image::images/repositories.png[{kib} Console,align="center"]
+
+. Get the details for the repository we'll recreate later by clicking on the repository
+name in the repositories table and making note of all the repository configurations
+that are displayed on the repository details page (we'll use them when we recreate
+the repository). Close the details page using the "X Close" link at
+the bottom left of the page.
```
Comment (dakrone) [inline docs/reference/tab-widgets/troubleshooting/snapshot/corrupt-repository.asciidoc:56]: I think taking out the text will help in the future if it ever changes (totally optional though)

```suggestion
the repository). Close the details page using the link at
the bottom left of the page.
```

File: docs/reference/tab-widgets/troubleshooting/snapshot/corrupt-repository.asciidoc [hunk 8]
```diff
@@ -0,0 +1,219 @@
...
++
+[role="screenshot"]
+image::images/repo_details.png[{kib} Console,align="center"]
+
+. With all the details above noted down, let's delete the repository. Select the
```
Comment (dakrone) [inline docs/reference/tab-widgets/troubleshooting/snapshot/corrupt-repository.asciidoc:61]:
```suggestion
. With all the details above noted, next delete the repository. Select the 
```

File: docs/reference/troubleshooting/snapshot/add-repository.asciidoc [hunk 9]
```diff
@@ -0,0 +1,11 @@
+[[add-repository]]
+== Multiple deployments writing to the same repository
```
Comment (dakrone) [inline docs/reference/troubleshooting/snapshot/add-repository.asciidoc:2]: Minor suggestion, up to you:

```suggestion
== Multiple deployments writing to the same snapshot repository
```

**Review by dakrone — APPROVED** [review]
LGTM, I left a few more nits, but up to you. Thanks for writing these!

**Comment by andreidan** [comment]
@elasticmachine update branch

**Pull request merged**